src/Rules/SyntaxErrorInQueryMethodRule.php

```diff
@@ -54,7 +54,7 @@ public function processNode(Node $node, Scope $scope): array
         foreach ($this->classMethods as $classMethod) {
             sscanf($classMethod, '%[^::]::%[^#]#%s', $className, $methodName, $queryArgPosition);
 
-            if ($methodName === $methodReflection->getName() && $className === $methodReflection->getDeclaringClass()->getName()) {
+            if ($methodName === $methodReflection->getName() && is_a($className, $methodReflection->getDeclaringClass()->getName(), true)) {
```

Comment:
I finally got the idea what you are trying to achieve here.

With `is_a` this check would require runtime reflection.
Instead it should use static reflection
https://github.com/phpstan/phpstan-src/blob/d5e6df75d42ad942f0cd1d592c0faf6a30c85c5d/src/Reflection/ClassReflection.php#L599